jmclient/jmclient/taker.py

```diff
@@ -647,8 +650,9 @@ def on_sig(self, nick, sigb64):
             # other guy sent a failed signature
 
         tx_signed = True
-        for ins in self.latest_tx.vin:
-            if ins.scriptSig == b"":
+        for input, witness in zip(self.latest_tx.vin, self.latest_tx.wit.vtxinwit):
+            if input.scriptSig == b"" \
+               and witness == btc.CTxInWitness(btc.CScriptWitness([])):
```

Comment:
Yes, this is the correct fixup to make bech32 work here. Note the `0xdeadbeef` markers will still be caught in the `scriptSig` check.